### PR TITLE
refactor: Break out farm fetch into own helper

### PR DESCRIPTION
**Pull request by hachiojidev**
Simply breaks out the fetching of a farm into a single function.

### CODE REVIEW

**Review by ChefHutch — APPROVED** [review]
Thanks for breaking these out into separate PRs 👍

**Comment by memoyil** [comment]
We should still apply the timeout I think otherwise It will not proceed. Wdyt?

**Review by Chef-Cheems — APPROVED** [review]

File: src/state/farms/fetchPublicFarmData.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import BigNumber from 'bignumber.js'
+import { Address, Token } from 'config/constants/types'
+import masterchefABI from 'config/abi/masterchef.json'
+import erc20 from 'config/abi/erc20.json'
+import { getAddress, getMasterChefAddress } from 'utils/addressHelpers'
+import { BIG_TEN } from 'utils/bigNumber'
+import multicall from 'utils/multicall'
+
+const fetchFarm = async (pid: number, lpAddresses: Address, token: Token, quoteToken: Token) => {
```
Comment (hachiojidev) [inline src/state/farms/fetchPublicFarmData.ts:9]: We have an issue with typing `FarmConfig`, `Farm` and the state in redux. To mitigate this I extracted the needed arguments for more accurate typing.

Comment (hachiojidev) [reply src/state/farms/fetchPublicFarmData.ts:9]: I also removed the merging of the config to keep this function more pure.

**Review by RabbitDoge — APPROVED** [review]

**Pull request merged**